testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,595 @@
+
+    def setUp(self):
+        self.tmpdir = tempdir.TempDir()
+        self.testfile = self.tmpdir.name + '/test.dcd'
+        self.readfile = DCD
+        self.natoms = 3341
+        self.expected_frames = 98
+        self.seek_frame = 91
+        self.expected_remarks = '''* DIMS ADK SEQUENCE FOR PORE PROGRAM                                            * WRITTEN BY LIZ DENNING (6.2008)                                               *  DATE:     6/ 6/ 8     17:23:56      CREATED BY USER: denniej0                '''
+
+        np.random.seed(1178083)
+        if self.charmm:
+            self.random_unitcells = np.random.random((self.expected_frames, 6)).astype(np.float64)
+        else:
+            self.random_unitcells = np.random.uniform(high=80,size=(self.expected_frames, 6)).astype(np.float64)
+
+        with DCDFile(self.readfile) as f_in, DCDFile(self.testfile, 'w') as f_out:
+            for index, frame in enumerate(f_in):
+                box=frame.unitcell.astype(np.float64)
+                f_out.write(xyz=frame.x,
+                            box=self.random_unitcells[index],
+                            step=f_in.istart,
+                            natoms=frame.x.shape[0],
+                            charmm=self.charmm, # DCD should be CHARMM
+                            time_step=f_in.delta,
+                            ts_between_saves=f_in.nsavc,
+                            remarks=f_in.remarks)
+
+    def tearDown(self):
+        try:
+            os.unlink(self.testfile)
+        except OSError:
+            pass
+        del self.tmpdir
+
+    def test_written_unit_cell_random(self):
+        # written unit cell dimensions should match for all frames
+        # using randomly generated unit cells but some processing
+        # of the cosine data stored in charmm format is needed
+        # as well as shuffling of the orders in the unitcell
+        # array based on the prcoessing performed by 
+        # DCDFile read and more generally relating to Issue 187
+        with DCDFile(self.testfile) as test:
+            curr_frame = 0
+            while curr_frame < test.n_frames:
+                written_unitcell = test.read()[1]
+                ref_unitcell = self.random_unitcells[curr_frame]
+                if self.charmm:
+                    ref_unitcell[1] = math.degrees(math.acos(ref_unitcell[1]))
+                    ref_unitcell[3] = math.degrees(math.acos(ref_unitcell[3]))
+                    ref_unitcell[4] = math.degrees(math.acos(ref_unitcell[4]))
+
+                    _ts_order = [0, 2, 5, 4, 3, 1]
+                    ref_unitcell = np.take(ref_unitcell, _ts_order)
+                curr_frame += 1
+                assert_allclose(written_unitcell, ref_unitcell,
+                                rtol=1e-05)
+
+
+class TestRandomUnitCellCHARMM(_TestDCDWriteRandom):
+    charmm = 1
+
+class TestRandomUnitCellNAMD(_TestDCDWriteRandom):
+    charmm = 0
```

Comment:
OK now I know why this tripped me up yesterday. In the unit-test you reshuffle the box dimensions compared to what we write. But you shouldn't need to do this. What should read exactly what we write without the need of shuffling the reference box dimensions.